src/components/EnterBarcode/EnterBarcodePage.tsx

```diff
@@ -28,17 +28,24 @@ const EnterBarcodePage = ({
       if (selectedPatient.length > 0) {
         // Found patient
         const {
-          collectionPointId: { id: patientCCPId },
+          collectionPointId: {
+            id: patientCCPId,
+            eventId: { id: patientEventId },
+          },
           id,
         } = selectedPatient[0];
         // Redirect to patient profile
-        history.replace(`/patients/edit/${patientCCPId}/${id}`);
+        history.replace(
+          `/events/${patientEventId}/ccps/${patientCCPId}/patients/${id}`
```

Comment:
not sure if you guys are experiencing the same issue, but when i get redirected to the patient profile page the fields don't always appear (this is probably out of scope for this ticket @petorsong )

Reply:
can you include a screenshot and steps to reproduce this issue? i haven't seen run into this but we can look into it if you provide more detail

Reply:
I have to comment out the last line 48 to get directed to this page and see this issue, and I'm the only one experiencing it 🤷‍♀️ I'm not sure this is an issue. 
![ezgif com-video-to-gif (1)](https://user-images.githubusercontent.com/38408276/91339877-1e2b5d80-e7a5-11ea-8224-d8a9d5060b2c.gif)

Reply:
hmm looks cache related issue and/or a byproduct of using `history.replace` over `history.push` maybe - either way looks like a bug

Reply:
It's an Apollo error but its easy to fix. We need to make sure that for <b>every</b> query that we perform that <b>all</b> nested objects are queried with the ID. This was not the case for the `GET_PATIENT_BY_ID` query; the event sub-object was not queried with its ID.